### ISSUE TITLE
Rediscover new tests

### ISSUE THREAD

**Issue by sysradium** [joeyespo/pytest-watch#44]
If I add a test file when PTW is already running the test is ignored by PTW unless I restart it. 

It seems that you listen on changes for a list of files you gathered on PTW start and do not look for the new ones. Haven't checked the code yet but I think it's so. Maybe we should discover new tests as they appear.

**Comment by joeyespo** [comment]
This should be working right now. If not, I'll be releasing a new version soon that'll be much more robust with re-running.

Some questions for you.

1. Where is your `test` directory in relation to your project? If `ptw` is run top-level, it should detect code changes and file additions/deletions in both your project and in your tests
2. What OS are you using?

Thanks!

**Comment by syradium** [comment]
It depends.

```
project_root/
        code.py
        tests/
             test_one.py
```

Then I run `ptw` from a project root.
Another setup when each module-folder has a `tests.py` file. But in this case I also run `ptw` from project's root.

In both cases that was Linux.

**Comment by joeyespo** [comment]
Ok, that all looks fine.

Just to be clear, if you save an existing test file, does ptw re-run?

**Comment by syradium** [comment]
As far as I remember - yes it does.

**Comment by joeyespo** [comment]
Just published v4.0, which includes a possible fix (2107da5d40eff3cc8602372591db7fa1c2233e42).

Try a `pip install --update pytest-watch` and feel free to re-open if the problem persists. Thanks for the issue, @sysradium!
